app/controllers/exceptions_controller.rb

```diff
@@ -0,0 +1,20 @@
+# custom exceptions controller to render error views
+class ExceptionsController < ApplicationController
+  layout 'application'
+
+  def render_error
+    @exception = request.env["action_dispatch.exception"]
+    MetricsService.report_error(@exception, request, current_user, @study)
+    ErrorTracker.report_exception(@exception, current_user, @study, params)
+    respond_to do |format|
+      format.html { render action: 'render_error', status: :internal_server_error }
+      format.json do
+        render json: {
```

Comment:
this method is very similar to the API method except for a couple of lines.  There might be room for a general `RequestUtils.handle_error(request, user, is_json)` method, or at least  extract the below lines out into a `RequestUtils.format_error_as_json(exception)` method so it's standardized between here and the API

Reply:
7f87f88